src/app/_components/BlockList/data/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { useQueryClient } from '@tanstack/react-query';
+
+import { BurnBlock } from '@stacks/blockchain-api-client';
+
+export function useBtcBlocksMap(btcBlocks: BurnBlock[]) {
+  const map = {} as Record<string, BurnBlock>;
+  btcBlocks.forEach(block => {
+    map[block.burn_block_hash] = block;
+  });
+  return map;
+}
+
+export function useRefetchInitialBlockList(queryKeys: string[][], callback: () => void) {
+  const queryClient = useQueryClient();
+
+  return async function (callback: () => void) {
+    await Promise.all(
+      queryKeys.map(queryKey => queryClient.invalidateQueries({ queryKey: queryKey }))
```

Comment:
True, it works. Technically I think just using refetch makes more sense here as invalidating a query in React Query marks the cached data as stale, which does not immediately trigger a refetch. Instead, React Query refetches the data the next time any component that uses this query is re-rendered or if the query's data is accessed again. 
Refetching queries OTH forcibly triggers a network request to update the data immediately, regardless of whether any component is currently displaying or using this data. It seems to make the most sense to use refetch here since the intent is to immediately fetch the data and update the component. Using invalidate query potentially could introduce some lag